### PR TITLE
Fixed #3003 - Consistent names for the TaskTracker interface

### LINKED ISSUE #3003: Consistent naming for TaskTracker and data methods
* getTasks => appTasks(appId)
* tasks => allTasks
* AppDataMap => TasksByApp

and some more. Also rename TaskTracker methods.

### PR DESCRIPTION
**Pull request by kolloch**
I used mostly IntelliJs renaming feature and then organize imports.
In some cases I had to change names manually.

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/MarathonScheduler.scala [hunk 1]
```diff
@@ -115,7 +115,7 @@ class MarathonScheduler @Inject() (
     log.warn(s"Error: $message\n" +
       s"In case Mesos does not allow registration with the current frameworkId, " +
       s"delete the ZooKeeper Node: ${config.zkPath}/state/framework:id\n" +
-      s"CAUTION: if you remove this node, all tasks started with the current frameworkId will be orphaned!")
+      s"CAUTION: if you remove this node, all allTasks started with the current frameworkId will be orphaned!")
```
Comment (meichstedt) [inline src/main/scala/mesosphere/marathon/MarathonScheduler.scala]: probably renamed by IDEA – please revert this change

Comment (kolloch) [inline src/main/scala/mesosphere/marathon/MarathonScheduler.scala]: yes, found more instances of this and correct this right now. Sorry.

**Comment by kolloch** [comment]
Hopefully found all instances of incorrectly changed strings and reverted them.

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 2]
```diff
@@ -529,8 +529,8 @@ class SchedulerActions(
       log.info(s"Scaling ${app.id} from $currentCount down to $targetCount instances")
       taskQueue.purge(app.id)
 
-      val toKill = taskTracker.getTasks(app.id).take(currentCount - targetCount)
-      log.info(s"Killing tasks: ${toKill.map(_.getId)}")
+      val toKill = taskTracker.appTasksSync(app.id).take(currentCount - targetCount)
+      log.info(s"Killing allTasks: ${toKill.map(_.getId)}")
```
Comment (meichstedt) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: another bad rename

**Comment by kolloch** [comment]
And found some more...

**Comment by meichstedt** [comment]
LGTM! Waiting for tests …
